### PR TITLE
Look for Celeste.exe inside Celeste.app/Contents/MacOS on Mac OS X

### PR DESCRIPTION
**Pull request by cdparks**
On Mac OS X (I'm using 10.14.6 Mojave), the `Celeste.exe` file is in `Celeste.app/Contents/MacOS` instead of `Celeste`. I've updated `findCelesteDir` to look there on Apple platforms.

It's somewhat important that we find the executable automatically since the native open dialogue doesn't let you navigate into `.app` bundles anymore.

### CODE REVIEW

**Comment by Cruor** [comment]
How does it behave in the case it ISN'T auto detecting a Steam version? Are you still able to select the the Celeste.exe in the .app manually?

**Comment by cdparks** [comment]
It pops this notification:

Screen Shot 2020-01-13 at 8 25 11 AM

Clicking `Close` opens the file dialogue. You used to be able to right-click app bundles and "See contents", but that functionality appears to have been disabled. Note that `Celeste.app` is greyed out:

Screen Shot 2020-01-13 at 8 25 30 AM

You can type `Command+Shift+G` to open the Goto-Location input:

Screen Shot 2020-01-13 at 8 25 51 AM

Which does let you look inside the app bundle if you paste the full path:

Screen Shot 2020-01-13 at 8 26 01 AM

**tl;dr** You _can_ get to the `exe`, but it's kind of a pain.

**Comment by cdparks** [comment]
@Cruor Is there anything special I need to do to get this merged? I'm not familiar with y'all's process

**Comment by Cruor** [comment]
Not really, I just forgot this PR exited. I'd rather have the full solution to the problem, because this hardly helps anyone in my opinion. It is still helpful, but what happens in the case that they don't use steam, considering the game has been free twice on Epic Game Store so far? 

I'd also rather have Lönn's Mac support tested, rather than actually looking into fixing this issue... Which reminds me I need to get the Mac build uploaded.

**Comment by Cruor** [comment]
All the Mac users (... like he two of them that exists) didn't seem to have problems with selecting the .app or using Steam though, I have never heard complaints about it. Not denying it wasn't a problem, but it is worth looking into,

**Comment by leo60228** [comment]
If I recall correctly, Catalina removed being able to browse inside a .app in a file picker.

**Comment by cdparks** [comment]
Maybe for Mac, the directive should be to browse for the `.app` instead of the `.exe`, and then we can find the `.exe` inside the `.app` once the user has selected it. That should be straightforward.

**Comment by cdparks** [comment]
I can update the PR to do something like that this weekend.

**Comment by Cruor** [comment]
Thats what Lönn attempts to do anyway, not that testers for Mac builds exist at all... Would appreciate if you wanted to test that. Feel free to also look at how the Mac file dialog stuff works, if it is helpful to you.
<https://github.com/CelestialCartographers/Loenn>

**Comment by cdparks** [comment]
Ah, ok, that looks like it does what I'd expect. I can test that this weekend.

**Comment by cdparks** [comment]
@Cruor, I tested the latest release of Lönn on my Mac, and it works fine. I wasn't sure how to build it from the GH repo though.

For Ahorn, I've done roughly the same thing - on Mac OS, the user is directed to select `Celeste.app` (and the file dialog is configured to filter `*.app` bundles). Once we have either `Celeste.exe` or `Celeste.app`, we can find the system-specific directory. Automatically finding the directory inside the Steam installation directory still works too.

**Comment by Cruor** [comment]
Sorry for late response. Looks good to me.
I'm going to take your word on it working on MacOS, I have nothing to test that on.
I'll merge it after testing it on Windows/Linux, just in case.

**Comment by cdparks** [comment]
Thanks, if you get other bug reports on MacOS, feel free to ping me. I'm happy to repro or investigate fixes.

**Comment by Vexatos** [comment]
I'll have to revert this, Cruor is an idiot. We have to locally merge this in our development repo first. It'll be actually merged with the next proper release.

**Comment by cdparks** [comment]
Huh. Do y'all cherry-pick commits into a develop branch or something?

**Comment by Vexatos** [comment]
More or less, yes.
